test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -168,6 +169,41 @@ object TestAspect extends TimeoutVariants {
       }
     }
 
+  /**
+   * An aspect that makes a test succeed if it fails for any cause.
+   */
+  val failure: PerTest[Nothing, Any, Nothing, Any, Unit, Unit] = failure(Assertion.anything)
+
+  /**
+   * An aspect that makes a test succeed if it fails for a given cause.
+   */
+  def failure[E0](p: Assertion[Cause[E0]]): PerTest[Nothing, Any, Nothing, E0, Unit, Unit] =
+    new TestAspect.PerTest[Nothing, Any, Nothing, E0, Unit, Unit] {
+      def perTest[R >: Nothing <: Any, E >: Nothing <: E0, S >: Unit <: Unit](
+        test: ZIO[R, TestFailure[E], TestSuccess[S]]
+      ): ZIO[R, TestFailure[E], TestSuccess[S]] = {
+        lazy val failValue: ZIO[R, TestFailure[E], TestSuccess[S]] = ZIO.fail(
+          TestFailure.Runtime(zio.Cause.die(new RuntimeException("expected failure")))
+        )
+
+        test.foldCauseM(
+          _.untraced match {
+            case Cause.Fail(TestFailure.Runtime(Traced(e, _))) if p.run(e).isSuccess =>
```

Comment:
`untraced` removes all inner `Traced` so this shouldn't be needed?

Reply:
from what I tested I had to get rid of it before I could match things.